### PR TITLE
Added install wizard

### LINKED ISSUE #74: install wizard
As a user, I'd want an option to check if I'm missing some software.
As a user, I'd want an option to install the missing software.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #74

### CODE REVIEW

File: src/commands/install.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import Vorpal from "vorpal";
+//import { execSync } from "child_process";
+//import { InstallUtils } from "../classes/install-utils";
```
Comment (eetupur) [inline src/commands/install.ts:3]: delete commented code

File: src/commands/install.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * software(string)
+ */
+async function action(args){
+  let software : string = args.options.software;
+  if(!software){
+    try {
+      const softwareResult = await this.prompt({
+        type : 'list',
+        name : 'software',
+        choices : ['NodeJs', 'GitHub', 'Git CLI', 'Maven', 'Java Development Kit 8', 'Java Development Kit 11', 'Homebrew'],
```
Comment (eetupur) [inline src/commands/install.ts]: This list will grow, recommend storing it as a variable in a less crowded place of the file (near the top perhaps)

**Review by eetupur — CHANGES_REQUESTED** [review]
Found a line to improve.

File: src/commands/install.ts (hunk 2 above)
Comment (veikkahonkanen) [reply src/commands/install.ts]: this is a placeholder either way, as these strings will be included in the types.ts as global constants in another issue

Comment (eetupur) [reply src/commands/install.ts]: that sounds fine

**Review by eetupur — APPROVED** [review]

File: src/commands/install.ts (hunk 1 above)
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
```

File: src/commands/install.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Installs software with a given flag, or activates a wizard to give options
+ * 
+ * @param args gets an object that can contain the software(string)'s name and options
+ * options include:
+ * software(string)
+ */
+async function action(args){
+  let software : string = args.options.software;
+  if(!software){
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
  let software : string = args.options.software;

  if(!software){
```

File: src/commands/install.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  }
+  this.log(`Attempting to install ${software}...`);
+  /*
+  const installCommand : string = await InstallUtils.installSW(software);
+  try {
+    execSync(installCommand);
+  } catch (err) {
+    throw new Error(`Error during ${software} installation process: ${err}`);
+  }
+  */
```
Comment (jarinykanen) [inline src/commands/install.ts:38]: Commented out?

File: src/commands/install.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+/**
+ * Exports contents of file to be usable by main.ts
+ * 
+ * @param vorpal vorpal instance
+ */
+export const installWizard = (vorpal : Vorpal) : Vorpal.Command => vorpal
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
export const installWizard = (vorpal: Vorpal): Vorpal.Command => vorpal
```

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/commands/install.ts (hunk 4 above)
Comment (veikkahonkanen) [reply src/commands/install.ts:38]: InstallUtils implemented in another, unmerged branch

**Review by eetupur — APPROVED** [review]

File: src/commands/install.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Installs software with a given flag, or activates a wizard to give options
+ * 
+ * @param args gets an object that can contain the software(string)'s name and options
+ * options include:
+ * software(string)
+ */
+async function action(args){
+  let software : string = args.options.software;
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
  let software: string = args.options.software;
```

File: src/commands/install.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+  if(!software){
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
  if (!software) {
```

File: src/commands/install.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    try {
+      const softwareResult = await this.prompt({
+        type : 'list',
+        name : 'software',
+        choices : ['NodeJs', 'GitHub', 'Git CLI', 'Maven', 'Java Development Kit 8', 'Java Development Kit 11', 'Homebrew', 'Docker', 'Minikube', 'Kubernetes CLI', 'Kustomize'],
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
        choices : [ 'NodeJs', 'GitHub', 'Git CLI', 'Maven', 'Java Development Kit 8', 'Java Development Kit 11', 'Homebrew', 'Docker', 'Minikube', 'Kubernetes CLI', 'Kustomize' ],
```

**Review by jarinykanen — APPROVED** [review]

File: src/commands/install.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import Vorpal from "vorpal";
+
+/**
+ * Installs software with a given flag, or activates a wizard to give options
+ * 
+ * @param args gets an object that can contain the software(string)'s name and options
+ * options include:
+ * software(string)
+ */
+async function action(args){
```
Comment (jarinykanen) [inline src/commands/install.ts]:
```suggestion
async function action(args) {
```

**Pull request merged**